internal/meta/meta_res.go

```diff
@@ -121,30 +128,93 @@ func (meta *MetaResource) ListResource(ctx context.Context) (ImportList, error)
 		}
 		l = append(l, item)
 	} else {
-		// Multi-resource mode only honors the resourceName[Pre|Suf]fix
-		for i, res := range rl {
-			tfAddr := tfaddr.TFAddr{
-				Type: "",
-				Name: fmt.Sprintf("%s%d%s", meta.resourceNamePrefix, i, meta.resourceNameSuffix),
-			}
-			item := ImportItem{
-				AzureResourceID: res.AzureId,
-				TFResourceId:    res.TFId,
-				TFAddr:          tfAddr,
-				TFAddrCache:     tfAddr,
-			}
-			if res.TFType != "" {
-				item.TFAddr.Type = res.TFType
-				item.TFAddrCache.Type = res.TFType
-				item.Recommendations = []string{res.TFType}
-				item.IsRecommended = true
-			}
-
-			l = append(l, item)
-		}
+		meta.appendRlToImportList(originalRl, &l)
 	}
 
+	meta.appendRlToImportList(extensionRl, &l)
+
 	l = meta.excludeImportList(l)
 
 	return l, nil
 }
+
+func (meta MetaResource) appendRlToImportList(rl []resourceset.TFResource, l *ImportList) {
+	for _, res := range rl {
+		tfAddr := tfaddr.TFAddr{
+			Type: "",
+			Name: fmt.Sprintf("%s%d%s", meta.resourceNamePrefix, len(*l), meta.resourceNameSuffix),
+		}
+		item := ImportItem{
+			AzureResourceID: res.AzureId,
+			TFResourceId:    res.TFId,
+			TFAddr:          tfAddr,
+			TFAddrCache:     tfAddr,
+		}
+		if res.TFType != "" {
+			item.TFAddr.Type = res.TFType
+			item.TFAddrCache.Type = res.TFType
+			item.Recommendations = []string{res.TFType}
+			item.IsRecommended = true
+		}
+
+		*l = append(*l, item)
+	}
+}
+
+func (meta MetaResource) queryResourceSet(ctx context.Context, resources []resourceset.AzureResource) (*resourceset.AzureResourceSet, error) {
+	var rl []resourceset.AzureResource
+
+	var ids []string
+	for _, res := range resources {
+		ids = append(ids, res.Id.String())
+	}
```

Comment:
Can we combine this loop with the one at L185, and put it right above where it was used (L188)?